Library/OcAppleKernelLib/KextPatcher.c

```diff
@@ -364,6 +364,103 @@ PatcherApplyGenericPatch (
   return EFI_NOT_FOUND;
 }
 
+EFI_STATUS
+PatcherExcludePrelinkedKext (
+  IN     CONST CHAR8            *Identifier,
+  IN OUT PATCHER_CONTEXT        *PatcherContext,
+  IN OUT PRELINKED_CONTEXT      *PrelinkedContext
+  )
+{
+  MACH_SEGMENT_COMMAND_ANY  *Segment;
+  VOID                      *KextData;
+  UINT64                    AddressMax;
+  UINT64                    VirtualAddress;
+  UINT64                    Size;
+  UINT64                    MaxSize;
+  UINT32                    KextCount;
+  UINT32                    Index;
+  UINT32                    Index2;
+  XML_NODE                  *KextPlist;
+  UINT32                    KextPlistCount;
+  CONST CHAR8               *KextPlistKey;
+  XML_NODE                  *KextPlistValue;
+  CONST CHAR8               *KextIdentifier;
+
+  ASSERT (Identifier       != NULL);
+  ASSERT (PatcherContext   != NULL);
+  ASSERT (PrelinkedContext != NULL);
+
+  //
+  // Zero out kext memory through PatcherContext->MachContext.
+  //
+  Segment    = NULL;
+  AddressMax = 0;
+  MaxSize    = 0;
+  while ((Segment = MachoGetNextSegment (&PatcherContext->MachContext, Segment)) != NULL) {
+    VirtualAddress = PatcherContext->Is32Bit ? Segment->Segment32.VirtualAddress : Segment->Segment64.VirtualAddress;
+    Size           = PatcherContext->Is32Bit ? Segment->Segment32.Size : Segment->Segment64.Size;
+
+    AddressMax = MAX (VirtualAddress + Size, AddressMax);
+    MaxSize    = AddressMax - PatcherContext->VirtualBase;
+  }
+
+  KextData = MachoGetFilePointerByAddress (
+    &PatcherContext->MachContext,
+    PatcherContext->VirtualBase,
+    NULL
+    );
+  if (KextData == NULL) {
+    return EFI_UNSUPPORTED;
+  }
+  DEBUG ((
+    DEBUG_INFO,
+    "OCAK: Excluding %a - VirtualBase %Lx, MaxSize %Lx\n",
+    Identifier,
+    PatcherContext->VirtualBase,
+    MaxSize
+    ));
+  ZeroMem (KextData, (UINTN) (AddressMax + MaxSize));
```

Comment:
It holds that
``MaxSize  == AddressMax - PatcherContext->VirtualBase``
So we have
``AddressMax + MaxSize == AddressMax + (AddressMax - PatcherContext->VirtualBase) == 2 * AddressMax - PatcherContext->VirtualBase``
Should this not be just ``MaxSize``?

Reply:
My bad, indeed!